eclipse-sirius/sirius-web
# [3371] Add support to specify the default visibility state of a node

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?
Open a studio and set the _Is Hidden By Default Expression_ and/or _Is Faded By Default Expression" of a NodeDescription to `aql:true`.
Create an element  matching the NodeDescription on a diagram, it should be hidden/faded as defined in the expressions.
